src/test.ts

```diff
@@ -54,18 +54,19 @@ export interface TestResult {
     };
 }
 
-export function runTests(patterns: string[], rulesDirectory?: string | string[]): TestResult[] {
+export async function runTests(patterns: string[], rulesDirectory?: string | string[]): Promise<TestResult[]> {
     const files: string[] = [];
     for (let pattern of patterns) {
         if (path.basename(pattern) !== "tslint.json") {
             pattern = path.join(pattern, "tslint.json");
         }
         files.push(...glob.sync(pattern));
     }
-    return files.map((directory: string): TestResult => runTest(path.dirname(directory), rulesDirectory));
+    return await Promise.all(
```

Comment:
Don't `return await` promises, just return the Promise. There are several cases thoughout this PR.

There's already a PR to add a rule to prevent this.

Reply:
Interesting! I remember anecdotally hearing some preference for `return await` for standardization of where execution context results during errors... but all I can find now to back it up is https://stackoverflow.com/questions/38708550/difference-between-return-await-promise-and-return-promise (and that doesn't). Will change.